### ISSUE TITLE
ORA-00904: tmp columns less target columns

### ISSUE THREAD

**Issue by AlexMorti** [oracle/dbt-oracle#26]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

if tmp relation columns are less than target relation columns we get an error ORA-00904.

need to get the columns from tmp_relation and not the target_relation
https://github.com/oracle/dbt-oracle/blob/ba7ace1b770b268caa1970c7412b7fdabf4acc50/dbt/include/oracle/macros/materializations/incremental/helpers.sql#L38

logs:
[dbt1.log](https://github.com/oracle/dbt-oracle/files/8934990/dbt1.log)
 

### Expected Behavior

08:02:37  1 of 1 START incremental model cdi_etl.customers................................ [RUN]
08:02:37  1 of 1 OK created incremental model cdi_etl.customers........................... [OK in 0.34s]
08:02:37  Finished running 1 incremental model in 1.23s.
08:02:37  Completed successfully

### Steps To Reproduce

CREATE TABLE customers(
    id NUMBER not null,
    first_name VARCHAR2(20) not null,
    last_name VARCHAR2(20) not null,
    hashvalue VARCHAR2(100)
);

model:
{{ config(
        materialized='incremental'
        )
}}
SELECT
98589 as id,
'alex' as first_name,
'morti' as last_name
FROM dual

run model:
08:00:00  1 of 1 START incremental model cdi_etl.customers................................ [RUN]
08:00:00  oracle adapter: Oracle error: ORA-00904: "HASHVALUE": invalid identifier
08:00:00  1 of 1 ERROR creating incremental model cdi_etl.customers....................... [ERROR in 0.29s]

### Environment

```markdown
- OS:macOS
- Python:3.9.7
- dbt_core: 1.0.8
- dbt_oracle:1.0.4
```

### What Oracle database version are you using dbt with?

12.1.0

**Comment by aosingh** [comment]
Hi @AlexMorti 

It is important to follow what dbt recommends and implements. dbt gets columns from the existing relation (target relation) and not the temporary relation. 

Below is the snippet of their default [incremental materialization implementation](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/include/global_project/macros/materializations/models/incremental/incremental.sql#L43-L48)
```
{% set dest_columns = process_schema_changes(on_schema_change, temp_relation, existing_relation) %}
{% if not dest_columns %}
       {% set dest_columns = adapter.get_columns_in_relation(existing_relation) %}
{% endif %}
```

For this issue, I recommend running dbt with `--full-refresh` mode 


In our next upcoming version`dbt-oracle==v1.1.0`, we support `on_schema_change` and you can simply use `on_schema_change=sync_all_columns` configuration without doing a full refresh. 

Note that `on_schema_change=sync_all_columns` will drop the column `hashvalue` from customers table (target) because your source (customer.sql) does not have `hashvalue` column

Let me know if you have any questions

I highly recommend reading dbt's documentation on [what-if-the-columns-of-my-incremental-model-change](https://docs.getdbt.com/docs/building-a-dbt-project/building-models/configuring-incremental-models#what-if-the-columns-of-my-incremental-model-change)

It explains 4 different scenarios. The possible values for `on_schema_change` are:

- `ignore`: Default behavior. dbt will get columns from existing relation. 
- `fail`: Triggers an error message when the source and target schemas diverge
- `append_new_columns`: Append new columns to the existing table. Note that this setting does not remove columns from the existing table that are not present in the new data.
- `sync_all_columns`: Adds any new columns to the existing table, and removes any columns that are now missing. Note that this is inclusive of data type changes.

**Comment by aosingh** [comment]
Hi @AlexMorti 

As mentioned in the above comment, the source and target schemas on this incremental model are out of sync

They can be reconciled in several ways:
  - set the `on_schema_change` config to either append_new_columns or sync_all_columns, depending on your situation.
  - Re-run the incremental model with `full_refresh: True` to update the target schema.
  - update the schema manually and re-run the process

Let me know if you have any questions.

Thanks

**Comment by aosingh** [comment]
@AlexMorti 

I am closing this issue for now. This was addressed in `dbt-oracle=1.1.0`

Please test it and let me know if you have any questions.
